python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py

```diff
@@ -454,7 +486,17 @@ def execute(self, context):
             self._run_id = context['dag_run'].run_id
         try:
             # FIXME implement intermediate result persistence to S3
-            return super(DagsterOperator, self).execute(context)
+
+            self.log.debug('Executing with query: {query}'.format(query=self.query))
+            raw_res = super(DagsterOperator, self).execute(context)
+            res = json.loads(raw_res)
+            if res['data']['startSubplanExecution']['hasFailures']:
+                raise AirflowException(
+                    res['data']['startSubplanExecution']['stepEvents'][0]['errorMessage']
+                )
+            return res
+        except JSONDecodeError:
```

Comment:
when does json error occur?

Reply:
added a comment, but this is if e.g. dagit crashes